spring-cloud-task-batch/src/main/java/org/springframework/cloud/task/batch/partition/DeployerPartitionHandler.java

```diff
@@ -169,6 +169,19 @@ public DeployerPartitionHandler(TaskLauncher taskLauncher, JobExplorer jobExplor
 		this.resource = resource;
 		this.stepName = stepName;
 		this.taskRepository = taskRepository;
+		this.threadPoolTaskExecutor = threadPoolTaskExecutor;
+	}
+
+	/**
+	 * Constructor initializing the DeployerPartitionHandler instance.
+	 * @param taskLauncher The {@link org.springframework.cloud.deployer.spi.task.TaskLauncher} used to execute partitioned tasks.
+	 * @param jobExplorer The {@link JobExplorer} to acquire the status of the job.
+	 * @param resource The {@link Resource} to the app to be launched.
+	 * @param stepName The name of the step.
+	 */
+	public DeployerPartitionHandler(org.springframework.cloud.deployer.spi.task.TaskLauncher taskLauncher, JobExplorer jobExplorer,
+		Resource resource, String stepName, TaskRepository taskRepository) {
+		this(taskLauncher, jobExplorer, resource, stepName, taskRepository, null);
```

Comment:
The default `TaskExecutor` could be a `SyncTaskExecutor` to preserve the current default behaviour of launching tasks sequentially.